sync/http/middleware.go

```diff
@@ -99,14 +100,33 @@ func NewAuthMiddleware(auth auth.Authenticator) MiddlewareFunc {
 	}
 }
 
-// NewTracingMiddleware creates a MiddlewareFunc that continues a tracing span and finishes it.
-func NewTracingMiddleware(path string) MiddlewareFunc {
+// NewLoggingTracingMiddleware creates a MiddlewareFunc that continues a tracing span and finishes it.
+// It also logs the HTTP request on debug logging level
+func NewLoggingTracingMiddleware(path string) MiddlewareFunc {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			sp, r := trace.HTTPSpan(path, r)
 			lw := newResponseWriter(w)
 			next.ServeHTTP(lw, r)
 			trace.FinishHTTPSpan(sp, lw.Status())
+			if log.Enabled(log.DebugLevel) {
```

Comment:
This function becomes bigger and bigger. can we extract the functionality to a new private function?